src/pages/api/revalidate.ts

```diff
@@ -0,0 +1,79 @@
+import { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+import { validateToken } from 'lib/api-validate-token'
+import { cachedGetProductData } from 'lib/get-product-data'
+
+/**
+ * Accepts a POST request with a product slug, triggers revalidation for all of a product's paths
+ * specified in its latest nav data.
+ */
+export default validateToken(
```

Comment:
⛏️I think this would be a smidge clearer if you defined the handler, and then exported the wrapped version. I spent a minute looking for where you were validating the token before I realized what was going on.

```ts
async function handler() {}

export default validateToken(handler, options)
```